src/Tactic/Rewrite.agda

```diff
@@ -0,0 +1,222 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- A simple tactic for used to automatically compute the function
+-- argument to cong.
+--
+-- The main use for this tactic is getting a similar experience to
+-- 'rewrite' during equational reasoning. This allows us to write very
+-- succinct proofs:
+--
+-- example : ∀ m n → m ≡ n → suc (suc (m + 0)) + m ≡ suc (suc n) + (n + 0)
+-- example m n eq = begin
+--     suc (suc (m + 0)) + m
+--   ≡⟨ rw (+-identityʳ m) ⟩
+--     suc (suc m) + m
+--   ≡⟨ rw eq ⟩
+--     suc (suc n) + n
+--   ≡˘⟨ rw (+-identityʳ n) ⟩
+--     suc (suc n) + (n + 0)
+--   ∎
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Tactic.Rewrite where
+
+open import Function using (_$_)
+
+open import Data.Bool.Base            using (true; false; if_then_else_; _∧_)
+open import Data.Char.Base   as Char  using (toℕ)
+open import Data.Float.Base  as Float using (_≡ᵇ_)
+open import Data.List.Base   as List  using ([]; _∷_)
+open import Data.Maybe.Base  as Maybe using (Maybe; just; nothing)
+open import Data.Nat.Base    as Nat   using (ℕ; zero; suc; _≡ᵇ_; _+_)
+open import Data.Unit.Base            using (⊤)
+open import Data.Word.Base   as Word  using (toℕ)
+open import Data.Product
+
+open import Relation.Binary.PropositionalEquality as Eq using (_≡_; refl; cong)
+
+-- 'Data.String.Properties' defines this via 'Dec', so let's use the builtin
+-- for maximum speed.
+import Agda.Builtin.String as String renaming (primStringEquality to _≡ᵇ_)
+
+open import Reflection
+open import Reflection.Abstraction
+open import Reflection.AlphaEquality        as Alpha
+open import Reflection.Argument             as Arg
+open import Reflection.Argument.Information as ArgInfo
+open import Reflection.Argument.Visibility  as Visibility
+open import Reflection.Meta                 as Meta
+open import Reflection.Name                 as Name
+open import Reflection.Term                 as Term
+
+open import Reflection.TypeChecking.Monad.Syntax
+
+----------------------------------------------------------------------
+-- Utilities
+----------------------------------------------------------------------
+
+private
+  -- Descend past a variable.
+  var-descend : ℕ → ℕ → ℕ
```

Comment:
This is going to be a bit of a pain I know, but the standard library uses camel case (`varDescend`) for function names, rather than kebab case (`var-descend`). Any chance you could fix the names in this file?